### DISCUSSION TITLE
Wildcard domain | redirect www subdomain

### DISCUSSION THREAD

**Discussion by andreyluchianic** [vercel/community#1047] [Help] [answered]
- domain uses Vercel nameservers
- in project domain is configured as `*.emergencyresponseguide.org`
- subdomain logic is on middleware level 

Problem: 
www.anysubdomain.emergencyresponseguide.org returns me `DEPLOYMENT_NOT_FOUND`. 
i can't configure in project->domain smth like `www.*.emergencyresponseguide.org`.

```bash
▶ curl -I https://texas.emergencyresponseguide.org -L
HTTP/2 200
age: 58
cache-control: public
content-type: text/html; charset=utf-8
date: Fri, 11 Nov 2022 13:03:43 GMT
etag: "ur3jymson2kn3"
server: Vercel
strict-transport-security: max-age=63072000
x-matched-path: /_sites/[subdomain]
x-powered-by: Next.js
x-vercel-cache: STALE
x-vercel-id: fra1::sfo1::qmts2-1668171881795-36f56625801d
content-length: 0
```

```
▶ curl -I https://www.texas.emergencyresponseguide.org -L
HTTP/2 404
cache-control: public, max-age=0, must-revalidate
content-type: text/plain; charset=utf-8
date: Fri, 11 Nov 2022 13:04:49 GMT
server: Vercel
strict-transport-security: max-age=63072000
x-vercel-cache: MISS
x-vercel-error: DEPLOYMENT_NOT_FOUND
x-vercel-id: fra1::kcgt2-1668171889472-aee4103d6628
content-length: 67
```

**Comment by andreyluchianic** [comment]
same here: 
https://www.testme.static.fun/
https://www.static.fun/

only root (no subdomain) works, cause i believe is separately configured rather then example in [documentation](https://vercel.com/blog/wildcard-domains) where just `*.static.fun` is configured

**Comment by amyegan** [comment]
Thanks for sharing these examples. I don't know exactly what happens during wildcard domain configuration, so I'm going to reach out to the engineering team for their input on this problem.

**Comment by smaeda-ks** [comment, marked as answer]
@andreyluchianic To be clear, in the case of `www.static.fun`, `www` IS a subdomain already, and this is not an Apex domain.
Wildcard only works for one level of subdomains (subdomains can't contain periods `.`). This is because TLS certificates won't support it either and you would still need to issue wildcard certs per your subdomain level.
